refactor(devtools): Improves error messaging for profiler file import

SkyZeroZx opened the pull request:
Formats error messages to display error name and message when importing profiler files fails
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

dgp1130 commented:
I've been trying to move us away from logging to the user visible console as much as possible, since it just adds noise to an output which the user is trying to parse for their own errors and issues. I think `console.log` in the DevTools panel context is generally ok, since that goes to its own console window.

That's really only for internal logging though. Any issues which should are user-visible should be presented in the UI rather than relying on the console.

SkyZeroZx committed:
refactor(devtools): Improves error messaging for profiler file import

Formats error messages to display error name and message when importing profiler files fails

SkyZeroZx commented:
The original console.error behavior is preserved; only a parsing step was added to make the error readable as a string in the Material snackbar.

crisbeto commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: aaa3020b94d168b7964f886b20f663487b627267
